### PR TITLE
Add text height diagram

### PR DESCRIPTION
**Pull request by GaryQian**
New simplified version that focuses on showing the relevant differences between the default font height and the em-square/font size height.

For https://github.com/flutter/flutter/pull/33281

This should be able to be landed cleanly before the height changes are in the engine.

### CODE REVIEW

File: packages/diagrams/lib/src/text.dart [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        text:' Aalg我हिन्दी ',
+        style: TextStyle(
+          fontSize: 100,
+          color: Colors.black,
+        ),
+      ),
+      textAlign: TextAlign.left,
+      textDirection: TextDirection.ltr,
+    );
+    switch (name) {
```
Comment (goderbauer) [inline packages/diagrams/lib/src/text.dart]: Why this switch?

File: packages/diagrams/lib/src/text.dart [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  const CurvePainter(this.textPainter);
+
+  final TextPainter textPainter;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    textPainter.layout();
+
+    final List<TextBox> boxes = textPainter.getBoxesForSelection(
+      const TextSelection(baseOffset: 0, extentOffset: 99)
```
Comment (goderbauer) [inline packages/diagrams/lib/src/text.dart]: What's the 99 magic number?

File: packages/diagrams/lib/src/text.dart [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      paint,
+    );
+
+    // emBottom
+    canvas.drawLine(
+      baseOffset + Offset(0, emBottom),
+      baseOffset + Offset(width, emBottom),
+      paint,
+    );
+
```
Comment (goderbauer) [inline packages/diagrams/lib/src/text.dart]: remove some of this extra white space?

File: packages/diagrams/lib/src/text.dart [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+      textAlign: TextAlign.left,
+      textDirection: TextDirection.ltr,
+    );
+    label.layout();
+    label.paint(canvas, baseOffset + Offset(0, baseline + 3));
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) {
+    return false;
```
Comment (goderbauer) [inline packages/diagrams/lib/src/text.dart]: Don't you want to repaint if the textPainters are different?

File: packages/diagrams/lib/src/text.dart [hunk 5]
```diff
@@ -0,0 +1,204 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```
Comment (goderbauer) [inline packages/diagrams/lib/src/text.dart]: group and separate imports per our style guide.

File: packages/diagrams/lib/src/text.dart (hunk 1 above)
Comment (GaryQian) [reply packages/diagrams/lib/src/text.dart]: This allows for adding additional diagrams using the same the same class in the future based off of a different name. It is not being used currently, so I can remove this for now if you want.

File: packages/diagrams/lib/src/text.dart (hunk 2 above)
Comment (GaryQian) [reply packages/diagrams/lib/src/text.dart]: It just needs to be a large number to ensure it extends out beyond the end of short string.

File: packages/diagrams/lib/src/text.dart (hunk 1 above)
Comment (goderbauer) [reply packages/diagrams/lib/src/text.dart]: I'd just remove it and add that when we actually need it.

**Pull request merged**